test/index.ts

```diff
@@ -46,40 +46,88 @@ import {
+    // @ts-expect-error cannot use undefined as dictionary key
+    Dictionary<number>[undefined],
+    // @ts-expect-error cannot use null as dictionary key
+    Dictionary<number>[null],
+    // @ts-expect-error cannot use string as only 'a' | 'b' allowed
+    Assert<IsExact<Dictionary<number, "a" | "b">[string], number>>,
+    Assert<IsExact<Dictionary<number, "a" | "b">["a"], number>>,
+    Assert<IsExact<Dictionary<number, "a" | "b">["b"], number>>,
+  ];
 }
 
 function testDictionaryValues() {
-  type Test = Assert<IsExact<DictionaryValues<Dictionary<number>>, number>>;
-}
-
-function testDictionaryValuesTwoArguments() {
-  type Test = Assert<IsExact<DictionaryValues<Dictionary<number, "a" | "b">>, number>>;
+  type cases = [
+    Assert<IsExact<DictionaryValues<Dictionary<string>>, string>>,
+    Assert<IsExact<DictionaryValues<Dictionary<number>>, number>>,
+    Assert<IsExact<DictionaryValues<Dictionary<boolean>>, boolean>>,
+    Assert<IsExact<DictionaryValues<Dictionary<bigint>>, bigint>>,
+    Assert<IsExact<DictionaryValues<Dictionary<symbol>>, symbol>>,
+    Assert<IsExact<DictionaryValues<Dictionary<undefined>>, undefined>>,
+    Assert<IsExact<DictionaryValues<Dictionary<null>>, null>>,
+    Assert<IsExact<DictionaryValues<Dictionary<string, "a" | "b">>, string>>,
+    Assert<IsExact<DictionaryValues<Dictionary<number, "a" | "b">>, number>>,
+    Assert<IsExact<DictionaryValues<Dictionary<boolean, "a" | "b">>, boolean>>,
+    Assert<IsExact<DictionaryValues<Dictionary<bigint, "a" | "b">>, bigint>>,
+    Assert<IsExact<DictionaryValues<Dictionary<symbol, "a" | "b">>, symbol>>,
+    Assert<IsExact<DictionaryValues<Dictionary<undefined, "a" | "b">>, undefined>>,
+    Assert<IsExact<DictionaryValues<Dictionary<null, "a" | "b">>, null>>,
+    Assert<IsExact<DictionaryValues<Dictionary<string, 1 | 2>>, string>>,
+    Assert<IsExact<DictionaryValues<Dictionary<number, 1 | 2>>, number>>,
+    Assert<IsExact<DictionaryValues<Dictionary<boolean, 1 | 2>>, boolean>>,
+    Assert<IsExact<DictionaryValues<Dictionary<bigint, 1 | 2>>, bigint>>,
+    Assert<IsExact<DictionaryValues<Dictionary<symbol, 1 | 2>>, symbol>>,
+    Assert<IsExact<DictionaryValues<Dictionary<undefined, 1 | 2>>, undefined>>,
+    Assert<IsExact<DictionaryValues<Dictionary<null, 1 | 2>>, null>>,
+    Assert<IsExact<DictionaryValues<SafeDictionary<string>>, string | undefined>>,
+    Assert<IsExact<DictionaryValues<SafeDictionary<number>>, number | undefined>>,
+    Assert<IsExact<DictionaryValues<SafeDictionary<boolean>>, boolean | undefined>>,
+    Assert<IsExact<DictionaryValues<SafeDictionary<bigint>>, bigint | undefined>>,
+    Assert<IsExact<DictionaryValues<SafeDictionary<symbol>>, symbol | undefined>>,
+    Assert<IsExact<DictionaryValues<SafeDictionary<undefined>>, undefined>>,
+    Assert<IsExact<DictionaryValues<SafeDictionary<null>>, null | undefined>>,
+    Assert<IsExact<DictionaryValues<SafeDictionary<string, "a" | "b">>, string | undefined>>,
```

Comment:
from this test case and below it's broken at the moment

I checked it in the broken commit – https://github.com/krzkaczor/ts-essentials/runs/3403457933

After it, it's fixed 🧪